discovery/db_discovery.go

```diff
@@ -24,33 +25,53 @@ var getTicker = func() *time.Ticker {
 	return time.NewTicker(cacheRefreshInterval)
 }
 
+type ticketParamsValidator interface {
+	ValidateTicketParams(ticketParams *pm.TicketParams) error
+}
+
+type roundsManager interface {
+	LastInitializedRound() *big.Int
+}
+
 type DBOrchestratorPoolCache struct {
-	node *core.LivepeerNode
+	store                 common.OrchestratorStore
+	lpEth                 eth.LivepeerEthClient
+	ticketParamsValidator ticketParamsValidator
+	rm                    roundsManager
+	bcast                 common.Broadcaster
 }
 
-func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode, rm roundsManager, ctx context.Context) (*DBOrchestratorPoolCache, error) {
```

Comment:
Let's make `ctx context.Context` the first parameter as that is convention (we don't use them right now, but static analysis tools that check for context propagation expect this convention - more details in the [docs](https://golang.org/pkg/context/)).

Reply:
fixed in d9c1a2d9